docs/DeveloperGuide.md

```diff
@@ -243,54 +244,250 @@ a. Include those libraries in the repo (this bloats the repo size)<br>
 b. Require developers to download those libraries manually (this creates extra work for developers)<br>
```

Comment:
Good job in describing the architectures... However, there are a few other things you can include such as you can explain how your design support undo/redo. There are multiple ways this can be done, so it will be useful to explain how you do it and why you chose to do it that way.